x/sequencer/keeper/msg_server_update_sequencer.go

```diff
@@ -4,36 +4,35 @@ import (
 	"context"
 	"fmt"
 
-	errorsmod "cosmossdk.io/errors"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
 	"github.com/dymensionxyz/dymension/v3/x/sequencer/types"
 )
 
-// UpdateSequencerInformation defines a method for creating a new sequencer
+// UpdateSequencerInformation defines a method for updating a sequencer
 func (k msgServer) UpdateSequencerInformation(goCtx context.Context, msg *types.MsgUpdateSequencerInformation) (*types.MsgUpdateSequencerInformationResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	rollapp, found := k.rollappKeeper.GetRollapp(ctx, msg.RollappId)
+	sequencer, found := k.GetSequencer(ctx, msg.Creator)
 	if !found {
-		return nil, types.ErrUnknownRollappID
+		return nil, types.ErrUnknownSequencer
 	}
 
-	if rollapp.Frozen {
-		return nil, types.ErrRollappJailed
+	if sequencer.Jailed {
+		return nil, types.ErrSequencerJailed
 	}
 
-	if err := msg.VMSpecificValidate(rollapp.VmType); err != nil {
-		return nil, errorsmod.Wrapf(types.ErrInvalidRequest, err.Error())
+	rollapp, found := k.rollappKeeper.GetRollapp(ctx, sequencer.RollappId)
```

Comment:
shouldnt it be MustGetRollapp? thinking if there is any situation where the sequencer has a rollapp id set and this rollapp doesn't exist. I think it's a broken invariant in that case.